### DISCUSSION TITLE
CSS Variable-based Themes: ETA?

### DISCUSSION THREAD

**Discussion by aofosev** [primefaces/community#1016] [PrimeReact] [answered]
From PrimeReact doc:

> Note: In near future, theming architecture of the styled mode will be redesigned to utilize CSS variables instead of SCSS variables in a backward compatible way for a dynamic approach.

This seems a very promising orientation (simplifying the toolchain, themes becoming simple sets of CSS variable assignments that can be changed dynamically -- and above all **allowing for "sub-themes"**, etc.)

Any update on a ballpark estimate of availability?

**Comment by melloware** [comment]
I think they were hoping for Q4 2024 or Q1 2025 but I have no personal insight.

**Comment by cagataycivici** [comment, marked as answer]
It is under heavy development, we plan to release a beta in November. PrimeVue is done, PrimeNG v18 is done, PrimeReact is the next one. Internally we built a new team for the next-gen version of PrimeReact, mainly consists of the folks who built PrimeVue led by our CTO. As you may know, we use the Vue version as the reference implementation and port it to Angular and React. 

Exciting times ahead for PrimeReact!
